File: Exercises/FTPTestApp/FTPTestApp/FTPState.vb
'http://msdn.microsoft.com/en-us/library/system.net.ftpwebrequest.aspx#Y4251

Imports System.Threading
Imports System.Net

Public Class FTPState
    Private wait As ManualResetEvent
    Private _request As FtpWebRequest
    Private _filename As String
    Private _operationException As Exception
    Private _statusDescription As String

    Sub New()
        wait = New ManualResetEvent(False)
    End Sub

    ReadOnly Property OperationComplete() As ManualResetEvent
        Get
            Return wait
        End Get
    End Property

    Property Request() As FtpWebRequest
        Get
            Return _request
        End Get
        Set(ByVal value As FtpWebRequest)
            _request = value
        End Set
    End Property

    Property Filename() As String
        Get
            Return _filename
        End Get
        Set(ByVal value As String)
            _filename = value
        End Set
    End Property

    Property OperationException() As Exception
        Get
            Return _operationException
        End Get
        Set(ByVal value As Exception)
            _operationException = value
        End Set
    End Property

    Property StatusDescription() As String
        Get
            Return _statusDescription
        End Get
        Set(ByVal value As String)
            _statusDescription = value
        End Set
    End Property
End Class

File: trunk/Exercises/FTPClient/FTPClient/FTPstate.cs
// Module:      FTPState.cs
//
// Description: modification of MSDN example for FTP put request
//              see below link for detail
// 
// Reference:   http://msdn.microsoft.com/en-us/library/system.net.ftpwebrequest.aspx#Y4251
// 
// Author:      MSDN
//              modified by CT Yeung
//
// Date:        18Sept12
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading;
using System.Net;

namespace FTPClient
{
    public class FtpState
    {
        private ManualResetEvent wait;
        private FtpWebRequest request;
        private string fileName;
        private Exception operationException = null;
        string status;

        public FtpState()
        {
            wait = new ManualResetEvent(false);
        }

        public ManualResetEvent OperationComplete
        {
            get { return wait; }
        }

        public FtpWebRequest Request
        {
            get { return request; }
            set { request = value; }
        }

        public string FileName
        {
            get { return fileName; }
            set { fileName = value; }
        }
        public Exception OperationException
        {
            get { return operationException; }
            set { operationException = value; }
        }
        public string StatusDescription
        {
            get { return status; }
            set { status = value; }
        }
    }
}
